lib/blockscore/connection.rb

```diff
@@ -0,0 +1,122 @@
+require 'json'
+require 'active_support/core_ext/module/attribute_accessors'
+require 'active_support/core_ext/string/inflections'
+require 'httparty'
+require 'blockscore/error'
+require 'blockscore/errors/invalid_request_error'
+
+HEADERS = {
+  'Accept' => 'application/vnd.blockscore+json;version=4',
+  'User-Agent' => 'blockscore-ruby/4.1.0 (https://github.com/BlockScore/blockscore-ruby)',
+  'Content-Type' => 'application/json'
+}
+
+module BlockScore
+  module Connection
+    mattr_accessor :api_key
+
+    def get(path, params)
+      request :get, path, params
+    end
+
+    def post(path, params)
+      request :post, path, params
+    end
+
+    def patch(path, params)
+      request :patch, path, params
+    end
+
+    def put(path, params)
+      request :put, path, params
+    end
+
+    def delete(path, params)
+      request :delete, path, params
+    end
+
+    private
+
+    def request(method, path, params)
+      response = execute_request(method, path, params)
+
+      case response.code
+      when 200, 201
+        json_obj = parse_json(response)
+
+        if json_obj.class == Array
+          build_response_from_arr json_obj
+        elsif json_obj[:object] == 'list'
+          build_response_from_arr json_obj[:data]
+        else
+          create_object(resource, json_obj)
+        end
+      else
+        handle_api_error(response.code, response.body)
+      end
+    end
+
+    def execute_request(method, path, params)
+      auth = { :username => @@api_key, :password => '' }
```

Comment:
If this were an internal PR I would say use `api_key` since that is why you defined the `mattr_accessor`.  With that said, I assume you will be ditching active_support in the final release, so maybe just remove that and make sure everything still behaves as expected.